Sources/Datadog/Logging/Log/LogEventBuilder.swift

```diff
@@ -48,8 +48,9 @@ internal struct LogEventBuilder {
         attributes: LogEvent.Attributes,
         tags: Set<String>,
         context: DatadogContext,
-        threadName: String
-    ) -> LogEvent? {
+        threadName: String,
+        callback: @escaping (LogEvent) -> Void
+    ) {
```

Comment:
We need to update the method comment too, notably this:
```swift
///
/// - Returns: the `LogEvent` or `nil` if the log was dropped by the user (from event mapper API).
///
```
is now replaced with asynchronous callback. Took me a while to understand that when the event is dropped, the callback is not called - worth putting it in the comment 👍.